src/pageEditor/modListingPanel/modals/CreateModModal.tsx

```diff
@@ -59,37 +61,51 @@ import { FieldDescriptions } from "@/modDefinitions/modDefinitionConstants";
 import useCreateModFromModComponent from "@/pageEditor/hooks/useCreateModFromModComponent";
 import useCreateModFromMod from "@/pageEditor/hooks/useCreateModFromMod";
 import { assertNotNullish, type Nullishable } from "@/utils/nullishUtils";
+import useCreateModFromUnsavedMod from "@/pageEditor/hooks/useCreateModFromUnsavedMod";
+import useIsMounted from "@/hooks/useIsMounted";
 
 function useInitialFormState({
   activeMod,
   activeModComponentFormState,
+  activeModId,
 }: {
   activeModComponentFormState: Nullishable<ModComponentFormState>;
   activeMod: ModDefinition | null;
+  activeModId?: RegistryId;
 }): ModMetadataFormState | UnknownObject {
   const scope = useSelector(selectScope);
   assertNotNullish(scope, "Expected scope to create new mod");
 
-  const activeModId =
-    activeModComponentFormState?.modMetadata?.id ?? activeMod?.metadata?.id;
+  // For unsaved mods, if the mod metadata has not been modified, it will only exist on the components
```

Comment:
```suggestion
  // For mods that have never been saved, the mod metadata will be present on the mod's respective mod components
```
Is this what you mean? What do you mean "if the mod metadata has not been modified" though? (i.e. if I have an unsaved mod and I've modified the mod metadata, what happens?)

Reply:
If you make changes on the mod metadata edit panel, it will create an entry in `editorSlice.dirtyModMetadataById`

Reply:
I've made some changes to clean up this component.

Reply:
> If you make changes on the mod metadata edit panel, it will create an entry in editorSlice.dirtyModMetadataById

Ah I see now